eth_abi/registry.py

```diff
@@ -0,0 +1,94 @@
+from .utils.parsing import collapse_type
+
+
+class ABIRegistryError(Exception):
+    pass
+
+
+class NoEntryFound(ABIRegistryError):
+    pass
+
+
+class PredicateMapping:
+    def __init__(self):
+        self._values = {}
+        self._labeled_predicates = {}
+
+    def add(self, predicate, value, label=None):
+        if label is not None:
+            self._labeled_predicates[label] = predicate
+
+        self._values[predicate] = value
```

Comment:
I think this should raise if the label or predicate already exist.

Reply:
Yep, thanks for the heads up.